src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -19,6 +19,7 @@
     <DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.1'">$(DefineConstants);EVENTSOURCE_ACTIVITY_SUPPORT</DefineConstants>
     <DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.1' and '$(TargetFramework)' != 'netstandard1.3'">$(DefineConstants);EVENTSOURCE_ENUMERATE_SUPPORT</DefineConstants>
     <DefineConstants Condition="$(TargetFramework.StartsWith('net4'))">$(DefineConstants);ALLOW_PARTIALLY_TRUSTED_CALLERS;ENABLE_HTTP_HANDLER</DefineConstants>
+    <DefineConstants Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">$(DefineConstants);NET5</DefineConstants>
```

Comment:
The usual pattern (both in this project and in runtime source I am more familiar with) is to name the defines based on the capability we want to enable. So this define would be named something like "W3C_DEFAULT_ID_FORMAT".

This has a few advantages:
1. Its a little more self-explanatory what the intent of the code is that falls within the define.
2. It separates the implementation of the feature from the policy choice about which versions/builds the feature should apply to. The policies often change over time even if the feature does not.

Reply:
I am not sure if this suggestion is better than what we have. we are not going to remove this code in the future, so pointing at enabling this feature (by the feature name) is not really going to help much. In coreclr code, we have spent a lot of time clearing such code and removing such ifdef's of feature names in the code because it became nonrelevant anyway. I chose NET5 to be explicit we have introduced this in .NET 5.0. I am not expecting in the future we'll enable/disable the feature anyway, so I am not sure what you mean by the `policy choice`, giving some example can help here.. 

Anyway, if you feel strongly about it, I'll change as it is not really worth the debate :-) please let me know what you think.

Reply:
I went ahead and changed the defined name as you have suggested.